### ISSUE TITLE
Dedicated Server Crash

### ISSUE THREAD

**Issue by katur239** [TeamPneumatic/pnc-repressurized#899]
*For feature requests, just erase this template and clearly describe the feature you'd like to see*

### Minecraft Version
1.165

### Forge Version
36.2.5

### Mod Version
2.14.3-257

### Describe your problem, including steps to reproduce it
Dedicate server crashing because a tick took longer than 60 seconds.

### Any other comments?
This is where it fails = [Server thread/DEBUG] [pneumaticcraft/]: Rebuilding Amadron offer list

Rolled back to 2.14.2-247 and problem went away.

Full error:

```
java.lang.Error: ServerHangWatchdog detected that a single server tick took 60.00 seconds (should be max 0.05)
	at com.terraforged.noise.util.Noise.cellEdge(Noise.java:250) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.noise.source.CellEdgeNoise.getValue(CellEdgeNoise.java:56) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.noise.source.NoiseSource.getValue(NoiseSource.java:62) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.noise.modifier.Warp.getValue(Warp.java:52) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.noise.modifier.Modifier.getValue(Modifier.java:49) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.noise.modifier.Modifier.getValue(Modifier.java:49) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.noise.modifier.Modifier.getValue(Modifier.java:49) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.module.Select.getSelect(Select.java:39) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.module.Blender.apply(Blender.java:69) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.world.continent.ContinentLerper2.apply(ContinentLerper2.java:63) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.world.heightmap.Heightmap.applyBase(Heightmap.java:196) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.world.heightmap.Heightmap.apply(Heightmap.java:181) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.world.heightmap.WorldLookup.compute(WorldLookup.java:108) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.engine.world.heightmap.WorldLookup.applyCell(WorldLookup.java:79) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.mod.biome.provider.TFBiomeProvider.computeBiome(TFBiomeProvider.java:213) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.mod.biome.provider.BiomeCache.computeValue(BiomeCache.java:117) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.mod.biome.provider.BiomeCache.getNoiseBiome(BiomeCache.java:81) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.mod.biome.provider.BiomeCache.tryGetBiome(BiomeCache.java:56) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.mod.biome.provider.TFBiomeProvider.lookupBiome(TFBiomeProvider.java:196) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at com.terraforged.mod.biome.provider.TFBiomeProvider.func_230321_a_(TFBiomeProvider.java:152) ~[terraforged:1.16.5-0.2.16-BETA-2] {re:classloading}
	at net.minecraft.world.server.ServerWorld.func_241116_a_(ServerWorld.java:1142) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:architects_palette.mixins.json:ServerWorldMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:APP:immersiveengineering.mixins.json:coremods.ServerWorldMixin,pl:mixin:APP:quark.mixins.json:ServerWorldMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at vazkii.quark.base.handler.MiscUtil.locateBiome(MiscUtil.java:238) ~[quark:r2.4-316] {re:classloading,pl:runtimedistcleaner:A}
	at vazkii.quark.content.tools.module.PathfinderMapsModule.createMap(PathfinderMapsModule.java:151) ~[quark:r2.4-316] {re:classloading}
	at vazkii.quark.content.tools.module.PathfinderMapsModule$PathfinderMapTrade.func_221182_a(PathfinderMapsModule.java:180) ~[quark:r2.4-316] {re:classloading}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager.lambda$null$10(AmadronOfferManager.java:330) ~[pneumaticcraft:1.16.5-2.14.3-257] {re:classloading}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager$$Lambda$35562/1183682256.accept(Unknown Source) ~[?:?] {}
	at java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:110) ~[?:1.8.0_292] {}
	at java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:581) ~[?:1.8.0_292] {}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager.lambda$null$11(AmadronOfferManager.java:328) ~[pneumaticcraft:1.16.5-2.14.3-257] {re:classloading}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager$$Lambda$35561/1376881494.accept(Unknown Source) ~[?:?] {}
	at java.util.Map.forEach(Map.java:630) ~[?:1.8.0_292] {}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager.lambda$setupVillagerTrades$12(AmadronOfferManager.java:327) ~[pneumaticcraft:1.16.5-2.14.3-257] {re:classloading}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager$$Lambda$35560/1710253077.accept(Unknown Source) ~[?:?] {}
	at java.util.HashMap.forEach(HashMap.java:1289) ~[?:1.8.0_292] {}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager.setupVillagerTrades(AmadronOfferManager.java:327) ~[pneumaticcraft:1.16.5-2.14.3-257] {re:classloading}
	at me.desht.pneumaticcraft.common.amadron.AmadronOfferManager.checkForFullRebuild(AmadronOfferManager.java:367) ~[pneumaticcraft:1.16.5-2.14.3-257] {re:classloading}
	at me.desht.pneumaticcraft.common.amadron.EventHandlerAmadron.amadronHousekeeping(EventHandlerAmadron.java:146) ~[pneumaticcraft:1.16.5-2.14.3-257] {re:classloading}
	at net.minecraftforge.eventbus.ASMEventHandler_565_EventHandlerAmadron_amadronHousekeeping_ServerTickEvent.invoke(.dynamic) ~[?:?] {}
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.eventbus.EventBus$$Lambda$2437/1708280164.invoke(Unknown Source) ~[?:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?] {}
	at net.minecraftforge.fml.hooks.BasicEventHooks.onPostServerTick(BasicEventHooks.java:120) ~[forge:?] {re:classloading}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:828) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:resourcefulbees.mixins.json:MixinMinecraftServer,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:resourcefulbees.mixins.json:MixinMinecraftServer,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:mixin:APP:betterendforge.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:mixin.terraforged.json:common.MixinMinecraftServer,pl:mixin:APP:resourcefulbees.mixins.json:MixinMinecraftServer,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.server.MinecraftServer$$Lambda$33869/1254532868.run(Unknown Source) ~[?:?] {}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292] {}
```

**Comment by desht** [comment]
There's some unexpected interaction here between Quark and/or Terraforged which I didn't see when made the related change.

Not sure who's at fault here, but I'll back out the change I made (which should have allowed trades involving maps to be added to Amadron, but I guess not...)

**Comment by desht** [comment]
Fixed in 2.14.4 release
